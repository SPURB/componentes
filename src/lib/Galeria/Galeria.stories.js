import { storiesOf } from '@storybook/vue'
import Galeria from './Galeria.vue'

storiesOf('Galeria', module)
  .add('background-size: 100% 100%',  () => ({
    components: { Galeria },
    template: `
      <Galeria
        :imagens="imagens"
        :backgroundCover="false"
        width="100%"
        height="300px"
      />
    `,
    data () {
      return {
        imagens: [
          'https://ddragon.leagueoflegends.com/cdn/img/champion/splash/Irelia_16.jpg', 
          'https://i.pinimg.com/originals/e1/b4/27/e1b42700f54e4db75f212c37c5589c7e.jpg'
        ]
      }
    }
  }))
  .add('bckground-size: cover', () => ({
    components: { Galeria },
    template: `
      <Galeria
        :imagens="imagens"
        :backgroundCover="true"
        width="100%"
        height="300px"
      />
    `,
    data () {
      return {
        imagens: [
          'https://www.dalcomad.com.br/wp-content/uploads/2018/11/capa.jpg', 
          'https://www.estudarfora.org.br/app/uploads/2019/04/urbanismo.jpg'
        ]
      }
    }
  }))
