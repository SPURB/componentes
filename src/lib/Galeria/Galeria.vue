<template>
  <div
    class="galeria"
    :style="{
      'background-image': `url(${imagens[currentImage]})`,
      'background-size': backgroundCover ? 'cover' : '100% 100%',
      'background-repeat': 'no-repeat',
      width,
      height
    }"
  >
    <slot name="header" />
    <section class="controles">
      <span 
        class="left"
        @click.prevent="previousDisabled ? '' : previous()"
      >
        <svg width="20" height="30" viewBox="0 0 248 400" fill="none" xmlns="http://www.w3.org/2000/svg">
          <path
            d="M247.667 353L95.0001 200L247.667 47L200.667 0L0.666748 200L200.667 400L247.667 353Z"
            :fill="previousDisabled ? '#ccc' : '#fff'"
          />
        </svg>
      </span>
      <span
        class="right"
        @click.prevent="nextDisabled ? '' : next()"
      >
        <svg width="20" height="30" viewBox="0 0 247 400" fill="none" xmlns="http://www.w3.org/2000/svg">
          <path
            d="M1.56015e-05 47L152.667 200L-1.11499e-05 353L47 400L247 200L47 -1.74846e-05L1.56015e-05 47Z"
            :fill="nextDisabled ? '#ccc' : '#fff'"
          />
        </svg>
      </span>
    </section>
  </div>
</template>

<script>
export default {
  name: 'Galeria',
  props: {
    imagens: {
      type: Array,
      required: true
    },
    width: {
      type: String,
      required: true
    },
    height: {
      type: String,
      required: true
    },
    backgroundCover: {
      type: Boolean,
      required: true
    }
  },
  data () {
    return {
      currentImage: 0
    }
  },
  computed: {
    lengthImagens () {
      return this.imagens.length
    },
    nextDisabled () {
      return this.lengthImagens - 1 == this.currentImage ? true : false
    },
    previousDisabled () {
      return this.currentImage == 0 ? true : false
    }
  },
  methods: {
    next () {
      this.currentImage = this.currentImage + 1
    },
    previous () {
      this.currentImage = this.currentImage - 1
    }
  }
}
</script>

<style scoped>
.galeria {
  display: flex;
  flex-direction: column;
  padding: 0 10px;
}

.galeria .controles {
  height: 100%;
  display: flex;
  align-items: center;
}

.galeria .controles span {
  cursor: pointer;
  transition: transform .15s ease-in-out;
  position: absolute;
}

.galeria .controles span.right {  
  right: 26px;
}

.galeria .controles span.right:hover {
  -moz-transform: translate(3px, 0px);
  -webkit-transform: translate(3px, 0px);
  -o-transform: translate(3px, 0px);
  -ms-transform: translate(3px, 0px);
  transform: translate(3px, 0px);
}

.galeria .controles span.left:hover {
  -moz-transform: translate(-3px, 0px);
  -webkit-transform: translate(-3px, 0px);
  -o-transform: translate(-3px, 0px);
  -ms-transform: translate(-3px, 0px);
  transform: translate(-3px, 0px);  
}
</style>