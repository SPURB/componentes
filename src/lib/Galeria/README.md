# Galeria.vue
Componente de galeria de imagens

## Exemplo
https://spurb.github.io/componentes/?path=/story/galeria--background-size-100-100

## Instruções de uso
Declare
```js
import { Galeria } from '@spurb/componentes'
export default {
  components: { Galeria }
}
```

Exemplo de uso no template
```html
<Galeria
  :imagens="[
    'https://www.clubesindicoprofissional.com.br/wp-content/uploads/SAO-PAULO11.jpg', 
    'https://www.estudarfora.org.br/app/uploads/2019/04/urbanismo.jpg'
  ]"
  :backgroundCover="true"
  width="100%"
  height="300px"
/>
```

### Props
| nome | data type | descrição |
|:-|:-|:-|
| `imagens` |  Array | Define as imagens da galeria |
| `backgroundCover` | Bool | Define se as imagens devem se ajustar a width e height passada |
| `width` | String | Define o width (largura) do Card, pode ser utilizado qualquer medida (px, %) |
| `height` | String | Define o height (altra) do Card, pode ser utilizado qualquer medida (px, %) |
