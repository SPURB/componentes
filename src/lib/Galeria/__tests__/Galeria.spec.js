import { shallowMount } from '@vue/test-utils'
import Galeria from '../Galeria.vue'

const imagens = [
  'https://www.dalcomad.com.br/wp-content/uploads/2018/11/capa.jpg',
  'https://www.estudarfora.org.br/app/uploads/2019/04/urbanismo.jpg'
]
const width = '100%'
const height = '300px'
const backgroundCover = false

describe('Galeria.vue', () => {
  it('Renderiza corretamente passando props', () => {
    const wrapper = shallowMount(Galeria, {
      propsData: { imagens, width, height, backgroundCover }
    })
    expect(wrapper.element).toMatchSnapshot() // compara renderização por snashot
  })
})
