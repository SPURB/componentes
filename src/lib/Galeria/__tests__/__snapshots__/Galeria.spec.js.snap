// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Galeria.vue Renderiza corretamente passando props 1`] = `
<div
  class="galeria"
  style="background-image: url(https://www.dalcomad.com.br/wp-content/uploads/2018/11/capa.jpg); background-size: 100% 100%; background-repeat: no-repeat; width: 100%; height: 300px;"
>
   
  <section
    class="controles"
  >
    <span
      class="left"
    >
      <svg
        fill="none"
        height="30"
        viewBox="0 0 248 400"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M247.667 353L95.0001 200L247.667 47L200.667 0L0.666748 200L200.667 400L247.667 353Z"
          fill="#ccc"
        />
      </svg>
    </span>
     
    <span
      class="right"
    >
      <svg
        fill="none"
        height="30"
        viewBox="0 0 247 400"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.56015e-05 47L152.667 200L-1.11499e-05 353L47 400L247 200L47 -1.74846e-05L1.56015e-05 47Z"
          fill="#fff"
        />
      </svg>
    </span>
  </section>
</div>
`;
